web/controllers/courseController/index.ts

```diff
@@ -2805,23 +2808,67 @@ async function groupMail(student: Student, courseId: number, subcourseId: number
         return 403;
     }
 
+    const lectures = subcourse.lectures.sort(
```

Comment:
By storing the return value of sort in a variable it seems as if `subcourse.lectures` is cloned, but in fact it is mutated. I think this would be clearer without the `lectures` local variable.

Reply:
Actually, as far as I can see a top-1 sort would be sufficient here: `const lastLecture = subcourse.lectures.reduce((a, b) => a.getMilliseconds() > b.getMilliseconds() ? a : b);`